### PR TITLE
Chip mbed ip integration

### PR DESCRIPTION
**Pull request by ATmobica**
#### Problem
Merge chip-mbed-ip intergration branch

 #### Summary of Changes
Mbed BSD socket implementation
Cypress board support
Lock-app application

### CODE REVIEW

File: src/platform/mbed/net/src/net_dns.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            res[i]          = new addrinfo;
+            res[i]->ai_addr = new sockaddr;
+            res[i]->ai_next = nullptr;
+            if (i > 0)
+            {
+                res[i - 1]->ai_next = res[i];
+            }
+            if (convert_mbed_addr_to_bsd(res[i]->ai_addr, &temp[i]) < 0)
+            {
+                // to  do free res  to avoid memory leak
```
Comment (pan-) [inline src/platform/mbed/net/src/net_dns.cpp:43]: Can it be implemented ?

File: src/platform/mbed/net/src/net_dns.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    return 0;
+}
+
+void mbed_freeaddrinfo(struct addrinfo * ai)
+{
+    struct addrinfo * next;
+    while (ai != NULL)
+    {
+        next = ai->ai_next;
+        free(ai);
```
Comment (pan-) [inline src/platform/mbed/net/src/net_dns.cpp:59]: `delete` must be used as the `addrinfo` is allocated with `new` and not `malloc`.
`ai->ai_addr` must be deleted too.

**Review by pan- — CHANGES_REQUESTED** [review]
I left few points if improvement that we can fix now. @ATmobica Can you take care of these then I will merge if there is no objections from you @lukgni @tymoteuszblochmobica

File: config/mbed/CMakeLists.txt [hunk 3]
```diff
@@ -157,6 +150,10 @@ if (CONFIG_CHIP_WITH_EXTERNAL_MBEDTLS)
     endif()
 endif()
 
+list(APPEND CHIP_CFLAGS
```
[the remark this answers is not in the document]
Comment (ATmobica) [reply config/mbed/CMakeLists.txt:153]: CHIP_CFLAGS is for GN propagation, CHIP_DEFINES is for cmake building - add to the main target.

File: src/inet/UDPEndPoint.cpp [hunk 4]
```diff
@@ -627,6 +631,13 @@ INET_ERROR UDPEndPoint::SendMsg(const IPPacketInfo * pktInfo, System::PacketBuff
     res = GetSocket(destAddr.Type());
     SuccessOrExit(res);
 
+    if (mState == kState_Ready)
+    {
+        mState = kState_Listening;
+        // Wake the thread calling select so that it starts selecting on the new socket.
+        lSystemLayer.WakeSelect();
```
[the remark this answers is not in the document]
Comment (ATmobica) [reply src/inet/UDPEndPoint.cpp:638]: This change is for UDP client implementation. I mentioned it once. We didn't call listen() function for the client. But when we send something we need to trigger select() and set a listening state to be ready to receive data from the server.

[the remark this answers is not in the document]
Comment (pan-) [reply src/inet/UDPEndPoint.cpp:638]: Thanks for the remainder.

**Review by pan- — APPROVED** [review]
LGTM

**Pull request merged**